static/js/compose_actions.js

```diff
@@ -72,32 +72,72 @@ export const _get_focus_area = get_focus_area;
 
 export function set_focus(msg_type, opts) {
     const focus_area = get_focus_area(msg_type, opts);
-    if (window.getSelection().toString() === "" || opts.trigger !== "message click") {
+    if (focus_area === "#compose-textarea" && opts.trigger === "toggle message") {
+        const $textarea = $("#compose-textarea");
+        // If the cursor is already in the compose box, keep it where it is.
+        // Otherwise, put it at the end of the compose box.
+        if (document.activeElement.id !== "compose-textarea") {
+            // Multiply by 2 to ensure the cursor always ends up at the end because
+            // Opera sometimes sees a carriage return as 2 characters.
+            const strLength = $textarea.val().length * 2;
+            $textarea.trigger("focus");
+            $textarea[0].setSelectionRange(strLength, strLength);
+        }
+    } else if (window.getSelection().toString() === "" || opts.trigger !== "message click") {
         const $elt = $(focus_area);
         $elt.trigger("focus").trigger("select");
     }
 }
 
+export const switch_to_private_message = $t({defaultMessage: "Switch to private message"});
+export const switch_to_stream_message = $t({defaultMessage: "Switch to stream message"});
```

Comment:
Let's add `_string` to the end of these variables -- at present they could sound a lot like the function you call to do the switch.